junit-platform-suite-api/src/main/java/org/junit/platform/suite/api/EnableImplicitConfigurationParameters.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2015-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.suite.api;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.apiguardian.api.API;
+import org.apiguardian.api.API.Status;
+
+/**
+ * Configure whether implicit configuration parameters should be considered.
+ *
+ * @see ConfigurationParameter
+ * @see Suite
+ * @see org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder#enableImplicitConfigurationParameters(boolean)
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Inherited
+@Documented
+@API(status = Status.EXPERIMENTAL, since = "1.8")
+public @interface EnableImplicitConfigurationParameters {
+
+	/**
+	 * By default implicit configuration parameters are considered.
+	 */
+	boolean enabled() default true;
```

Comment:
This is not consistent with `UseTechnicalNames` which are not used by default. The alternative would have been to create a `DisableImplicitConfigurationParameters` instead. However  `@UseTechnicalNames` is specific to the `JUnitPlatformRunner` only and I think it is better to mirror the discovery api like all other annotations do.

Reply:
I have not reviewed this PR other than this comment, but I would like to point out that an annotation with a single attribute with a default value (that represents the default behavior) is typically not something we would introduce since the annotation can be declared without any attributes and would zero effect on the behavior.

Normally we would introduce a marker annotation for such a purpose, something like `@DisableImplicitConfigurationParameters` that declares no attributes.

Of course, if it's foreseeable that one may later need to configure other behaviors related to configuration parameters, `@ConfigurationParameters` with use case specific attributes might be a better choice -- for example, `@ConfigurationParameters(disableImplicitParameters = true)`.

Reply:
Cheers. I do have an other use case in mind that is related to suite configuration parameters. I'll add that to the MR so we have something concrete to discuss.